lib/elixir/lib/system.ex

```diff
@@ -234,6 +237,7 @@ defmodule System do
 
   Returns the current working directory or raises `RuntimeError`.
   """
+  @spec cwd!() :: String.t() | no_return
```

Comment:
We don't need `no_return`. `no_return` is always implicit. We only use `no_return` when the only option is `no_return`. One way to look at it is to consider `no_return` as an equivalent to `none`. So `something | none` will always be `something`. :)

Reply:
For reference for these `!` functions I looked at how the types were defined in `File` for the `!` functions there, and they all use `:ok | no_return` (like [here](https://github.com/elixir-lang/elixir/blob/1cce731a363a92c494871af1823f935e48a790b5/lib/elixir/lib/file.ex#L236)). To be consistent, should we remove `no_return` in types with that signature and only use `no_return` in places where it's the only possible outcome?

Reply:
I would use `no_return` only in the places where it's the only possible outcome.

One could argue that `something | no_return` could specify that an error/exception is expected. However, I think the documentation and the naming convention of using a `!` should be enough :)

Reply:
@devonestes yes, those uses of `| no_return` is wrong and we should/could remove them.